deploy.js

```diff
@@ -0,0 +1,139 @@
+// Build directory.
+const BUILD_DIR = process.env.BUILD_DIR || "./dist";
+
+// Resolver datakey, e.g. "skynet-mysky". Allowed to be empty.
+const RESOLVER_DATA_KEY = process.env.RESOLVER_DATA_KEY || "";
+
+// Seed for generating and updating resolver skylink. Not allowed to be empty.
+const RESOLVER_SEED = process.env.RESOLVER_SEED || "";
+
+// A space-separated list of files that the portal should try when resolving a directory.
+const TRY_FILES = process.env.TRY_FILES || "index.html";
+
+// Defines a path to a file that will replace the default 404 Not Found error page, ie `404.html`.
+const NOT_FOUND_PAGE = process.env.NOT_FOUND_PAGE;
+
+// ===============================
+// ===============================
+// ===============================
+
+// Create client for upload and resolver skylink.
+const client = new SkynetClient(PORTAL);
+
+/**
+ * Uploads the directory at the path.
+ *
+ * @param path - The directory path.
+ * @returns - Returns the upload directory response.
+ */
+async function pushDirectoryToSkynet(path) {
+  return await client.uploadDirectory(path, prepareUploadOptions());
+}
+
+/**
+ * Publishes the skylink to a resolver skylink.
+ *
+ * @param skylink - The data link to publish.
+ * @param resolverSeed - The seed.
+ * @param resolverDataKey - The data key.
+ * @returns - The resolver skylink.
+ */
+async function publishSkylinkToResolverSkylink(skylink, resolverSeed, resolverDataKey) {
+  // Setup Keys for Read/Write of Mutable Data
+  const { privateKey, publicKey } = genKeyPairFromSeed(resolverSeed);
+  const dataKey = resolverDataKey;
+
+  // Set Registry Entry to point at our Skylink
+  await client.db.setDataLink(privateKey, dataKey, skylink);
+
+  // Get the resolver skylink that represents the registry entry
+  return await client.registry.getEntryLink(publicKey, dataKey);
+}
+
+/**
+ * Deploys the build directory.
+ *
+ * @returns - An empty promise.
+ */
+async function deploy() {
+  if (!RESOLVER_SEED) {
+    throw new Error("RESOLVER_SEED env var not found");
+  }
+
+  console.log("Sending to Skynet...");
+  const skylink = await pushDirectoryToSkynet(BUILD_DIR);
+  let resolverSkylinkUrl = "";
+
+  if (!skylink) {
+    throw new Error("App deployment failed");
+  }
+
+  // Get URL based off preferred portal
+  const skylinkUrl = await client.getSkylinkUrl(skylink, { subdomain: true });
+
+  console.log(`📡 App deployed to Skynet with skylink: ${skylink}`);
+
+  console.log();
+
+  // Call method to update resolver skylink.
+  const resolverSkylink = await publishSkylinkToResolverSkylink(skylink, RESOLVER_SEED, RESOLVER_DATA_KEY);
+
+  // Get URL based off preferred portal
+  resolverSkylinkUrl = await client.getSkylinkUrl(resolverSkylink, { subdomain: true });
+
+  console.log(`📡 Resolver skylink updated: ${resolverSkylink}`);
+
+  // Display final info.
+  console.log("🚀 Deployment to Skynet complete!");
+  console.log();
+  console.log(`Use the link${resolverSkylinkUrl && "s"} below to access your app:`);
```

Comment:
Why do we have this conditional "s"? Do we expect to not have the `resolverSkylinkUrl` in some case? If that's the case we should check for it when outputting it with a `console.log` later on (line 100). But it seems that if we fail to get/update the resolver link we'll throw an exception anyway.